python/tests/kat/t_grpc_stats.py

```diff
@@ -66,14 +66,14 @@ def check(self):
         stats = self.results[-1].text
 
         metrics = [
-            'envoy_cluster_grpc_EchoService_0',
-            'envoy_cluster_grpc_EchoService_13',
-            'envoy_cluster_grpc_EchoService_request_message_count',
-            'envoy_cluster_grpc_EchoService_response_message_count',
-            'envoy_cluster_grpc_EchoService_success',
-            'envoy_cluster_grpc_EchoService_total',
+            'envoy_cluster_grpc_Echo_0',
+            'envoy_cluster_grpc_Echo_13',
+            'envoy_cluster_grpc_Echo_request_message_count',
+            'envoy_cluster_grpc_Echo_response_message_count',
+            'envoy_cluster_grpc_Echo_success',
+            'envoy_cluster_grpc_Echo_total',
             # present only when enable_upstream_stats is true
-            'envoy_cluster_grpc_EchoService_upstream_rq_time'
```

Comment:
If stats metrics names have changed, then this probably warrants a mention in the release notes?

Reply:
As far as I can tell this is the only thing that has changed: <https://www.envoyproxy.io/docs/envoy/latest/version_history/v1.18/v1.18.0#april-15-2021>. I don't think that change effects this test.

In the docs it says: https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/grpc_stats_filter#config-http-filters-grpc-stats

>This filter enables telemetry of gRPC calls. It counts the number of successful and failed calls, optionally grouping them by the gRPC method name.

Which matches the naming of `envoy_cluster_grpc_Echo_*`... I also don't see anywhere, where our code changed. So, I will investigate more after getting this green.

Reply:
Ok, based on your awesome investigation we have come to find out this is in fact due to behavior change between Envoy 1.17 and 1.18. I have gone ahead and updated the change log to include a note about this and we will need to address it in upstream envoy based on your discoveries.